src/hotspot/share/runtime/sharedRuntime.cpp

```diff
@@ -447,6 +447,86 @@ JRT_LEAF(jdouble, SharedRuntime::l2d(jlong x))
   return (jdouble)x;
 JRT_END
 
+// Reference implementation at src/java.base/share/classes/java/lang/Float.java:floatToFloat16
+JRT_LEAF(jshort, SharedRuntime::f2hf(jfloat  x))
+  jint doppel = SharedRuntime::f2i(x);
```

Comment:
Newly added constant value computation runtime routines can be validated by a gtest.

Reply:
I have created a bug entry for the gtest. Will add it in a separate PR.